lib/manageiq/messaging/client.rb

```diff
@@ -40,6 +40,17 @@ class Client
       def self.open(options)
         protocol = options[:protocol] || :Stomp
         client = Object.const_get("ManageIQ::Messaging::#{protocol}::Client").new(options)
+
+        prepend_handler("INT") do |old|
+          client.close
+          old.call
+        end
+
+        prepend_handler("TERM") do |old|
+          client.close
+          old.call
+        end
```

Comment:
@bzwei is this needed because of the switch to rdkafka or unrelated?  Seems like overkill when the caller could just put `client.close` in an `ensure` block and I don't know that I want a gem messing with my signal handlers.  What happens if the caller called `.close` already?

Reply:
Not specifically for rdkafka. When we have a thread subscribe to a topic, the thread normally runs until the app shuts down. This code ensures the connection closes properly if the caller does not have a chance to do so. It is safe to call `.close` multiple times.

Because we do `old.call`,  existing signal handler is not overridden.

Reply:
Yeah I realize it isn't overridden but seems like extra complexity when a simple `ensure` would do the same thing.  Anyway I won't hold this up for this just seems odd

Reply:
Actually an `ensure` would be better because this only cleans up the connection when the process exist where an ensure at the end of the method would close it immediately, e.g.

```
def worker_thread
  loop do
    client = ManageIQ::Messaging::Client.open
    client.subscribe_topic do
    end
  rescue => err
    # log an error and retry
  end
end
```

Would open _lots_ of connections without closing them

Reply:
It is the responsibility of the caller who uses messaging gem to do `ensure`. Capturing `TERM` is an additional helper offered by this gem. But you're right. It should not be offered by a gem. I'll remove it.